### DISCUSSION TITLE
issue: unregister does not work correctly when conditionally rendering fields with the same key

### DISCUSSION THREAD

**Discussion by archiemcquaid** [react-hook-form/react-hook-form#11736] [Q&A]
### Version Number

7.51.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/eloquent-lake-y6pm3d

### Steps to reproduce

Hi. I have an issue where values from unregistered fields persist if another field is registered with the same key afterwards.

1. Country is set to UK. This renders the Province input with `name="province"`.
2. Enter `test` in the Province input
3. Select USA from country select. This triggers the province input field to unmount and `unregister` is called. A state select is rendered with the same `name="province"`
4. Click submit
5. `onSubmit` is called with `{country: 'US', province: 'test'}`

Interestingly when logging the result of `watch()` it shows `province` is correctly removed however when clicking `onSubmit`, the old value returns. 

<img width="343" alt="image" src="https://github.com/react-hook-form/react-hook-form/assets/16037240/b5703eea-8b47-4475-bcaf-ca2d95aeed7c">

### Expected behaviour

1. Country is set to UK. This renders the Province input with `name="province"`.
2. Enter `test` in the Province input
3. Select USA from country select. This triggers the province input field to unmount and `unregister` is called. A state select is rendered with the same `name="province"`
4. Click submit
5. Validation fails as `province` is unset and no state has been selected.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
https://react-hook-form.com/faqs#Whyisdefaultvaluenotchangingcorrectlywithternaryoperator
